modules/ref_arch_setup/tasks/install_pe.sh

```diff
@@ -26,6 +25,76 @@ execute_command() {
     fi
 }
 
+# Sometimes it takes more than one run ...
+
+execute_puppet_until_idempotent() {
+  puppet_command="puppet agent -t"
+
+  # The following could be converted into parameters with defaults ...
+
+  maximum_number_of_puppet_runs=8
+  sleep_seconds_when_puppet_run_in_progress=16
+  retry_when_success_with_failures=false
+
+  for run in $(seq 1 $maximum_number_of_puppet_runs); do
+    echo "Executing: '$puppet_command'"
+
+    # To both capture and output command output ...
+    # exec 5>&1
+    # puppet_command_output=$($puppet_command | tee /dev/fd/5; exit ${PIPESTATUS[0]})
+
+    puppet_command_output=$($puppet_command)
```

Comment:
^^ This captures the output of `puppet agent -t' where previously it output to STDOUT. If we want output to STDOUT, we should use Lines 43-44 instead of Line 46.

Reply:
We do want that.